packages/invoices/src/OutboundShipment/DetailView/modals/ItemDetailsModal.tsx

```diff
@@ -1,435 +1,124 @@
-    if (defaultPackSize.value && typeof defaultPackSize.value == 'number') {
-      setPackSize(defaultPackSize.value);
-    }
-    if (packSizes.length === 0) {
-      setSelected({ label: '', value: '' });
-    }
-  }, [defaultPackSize.value]);
-
-  return { selected, setPackSize, options };
-};
-
-const sumAvailableQuantity = (batchRows: BatchRow[]) => {
-  const sum = batchRows.reduce(
-    (acc, { availableNumberOfPacks, packSize }) =>
-      acc + availableNumberOfPacks * packSize,
-    0
-  );
-
-  return sum;
-};
-
-const getAllocatedQuantity = (batchRows: BatchRow[]) => {
-  return batchRows.reduce(
-    (acc, { numberOfPacks, packSize }) => acc + numberOfPacks * packSize,
-    0
-  );
-};
-
-const issueStock = (
-  batchRows: BatchRow[],
-  idToIssue: string,
-  value: number
-) => {
-  const foundRowIdx = batchRows.findIndex(({ id }) => id === idToIssue);
-  const foundRow = batchRows[foundRowIdx];
-  if (!foundRow) return [];
-
-  const newBatchRows = [...batchRows];
-  newBatchRows[foundRowIdx] = {
-    ...foundRow,
-    numberOfPacks: value,
-  };
-
-  return newBatchRows;
-};
-
 export const ItemDetailsModal: React.FC<ItemDetailsModalProps> = ({
   isOpen,
   onClose,
-  upsertInvoiceLine,
-  onChangeItem,
-  summaryItem,
-  onNext,
-  isEditMode,
-  isOnlyItem,
-  draft,
+  item,
+  mode,
 }) => {
+  const [currentItem, setCurrentItem] = useBufferState(item);
   const t = useTranslation(['distribution']);
-  const methods = useForm({ mode: 'onBlur' });
-  const { reset, register } = methods;
-
-  const { batchRows, setBatchRows, isLoading } = useBatchRows(summaryItem);
-  const packSizeController = usePackSizeController(batchRows);
-
-  const { hideDialog, showDialog, Modal } = useDialog({
-    onClose,
-  });
-
-  const onReset = () => {
-    reset();
-  };
-  const onCancel = () => {
-    onClose();
-    onReset();
-  };
-  const upsert = () => {
-    if (!summaryItem) return null;
-
-    // TODO: Handle placeholder upserting.
-    const invoiceLines = batchRows
-      .filter(({ id }) => id !== 'placeholder')
-      .map(batch =>
-        getInvoiceLine(generateUUID(), summaryItem, batch, batch.numberOfPacks)
-      );
-
-    // Upsert each line. Any lines which do no already exist and have no had any
-    // packs allocated, will not be created, but those which already exist and have
-    // their quantity reduced to 0 will be marked for deletion.
-    invoiceLines.forEach(upsertInvoiceLine);
-
-    onReset();
-  };
-  const upsertAndClose = () => {
-    upsert();
-    onClose();
-    onReset();
-    onChangeItem(null);
+  const { mutate } = useSaveOutboundLines();
+  const { status } = useOutboundFields('status');
+  const {
+    draftOutboundLines,
+    updateQuantity,
+    setDraftOutboundLines,
+    isLoading,
+  } = useDraftOutboundLines(item);
+  const packSizeController = usePackSizeController(draftOutboundLines);
+  const { Modal } = useDialog({ isOpen, onClose });
+  const nextItem = useNextItem();
```

Comment:
This is all *very* similar to the 'main' edit modal piece in the other packages.

Reply:
```suggestion
  const nextItem = useNextItem(item?.id);
```

clicking on the `next` button is just showing a new item view